### DISCUSSION TITLE
AngularFireAuth's currentUser in ngOnInit

### DISCUSSION THREAD

**Discussion by zgosalvez** [angular/angularfire#2733] [Q&A] [answered]
On an older version, I could easily get the `currentUser` synchronously inside `ngOnInit()`. This is very useful when I need the `currentUser.uid` of the currently logged in user. However, after upgrading, `currentUser` now returns a promise instead, breaking my `ngOnInit()` and subcomponent. **What's the best/easiest way to refactor my code to make this work again?**

**Comment by mattbodman** [comment, marked as answer]
As far as I can tell, you can add `async` in front of your `ngOnInit()` function.  Otherwise, you could do `auth.currentUser.then(...)`

**Reply by zgosalvez** [reply]
Decided on this route as well. Forgot to update this post. So far I haven't had any problems with this implementation.
